control/protocol.yaml

```diff
@@ -0,0 +1,102 @@
+openapi: 3.0.3
+info:
+  title: Control protocol models
+  description: >-
+    All control protocol schemas
+  version: ^0.0.0
+components:
+  schemas:
+    State.Protocol.All:
+      description: >-
+        Sets all configured protocols to `start` or `stop` state.
+        
+        Setting protocol state to `start` shall be a no-op if preceding `set_config` API call
+        was made with `config.options.protocol_options.auto_start_all` set to `true` or if
+        all the configured protocols are already started.
+      type: object
+      required:
+        - state
+      properties:
+        state:
+          description: >-
+            Protocol states
+          type: string
+          x-field-uid: 1
+          x-enum:
+            start:
+              x-field-uid: 1
+            stop:
+              x-field-uid: 2
+    State.Protocol.Route:
+      description: >-
+        Sets the state of configured routes
+      type: object
+      required:
+        - state
+      properties:
+        names:
+          description: >-
+            The names of device route objects to control.
+            If no names are specified then all route objects that match the
+            x-constraint will be affected.
+          type: array
+          items:
+            type: string
+          x-constraint:
+          - '/components/schemas/Bgp.V4RouteRange/properties/name'
+          - '/components/schemas/Bgp.V6RouteRange/properties/name'
+          - '/components/schemas/Isis.V4RouteRange/properties/name'
+          - '/components/schemas/Isis.V6RouteRange/properties/name'
+          x-field-uid: 1
+        state:
+          description: >-
+            Route states
+          type: string
+          x-field-uid: 2
+          x-enum:
+            withdraw:
+              x-field-uid: 1
+            advertise:
+              x-field-uid: 2
+    State.Protocol.Lacp:
+      description: >-
+        Sets state of configured LACP
+      type: object
+      required:
+        - choice
+      properties:
+        choice:
+          type: string
+          x-enum:
```

Comment:
Will we have any other option besides `admin` in the future?  I mean do we need the choice here or can we just expose the properties here?

Reply:
Yes, we could have `session` state which is different from `admin` state. In general, we'll end up having a separate schema object for every protocol which in-turn shall encapsulate different kinds of operational states.